### PR TITLE
Avoid undefined jinja errors for skipped recipes

### PR DESCRIPTION
**Pull request by msarahan**
### Description

A few NVIDIA feedstocks on conda-forge have been failing to re-render lately. See https://github.com/conda-forge/cuda-nvdisasm-feedstock/issues/12. The expected behavior was that undefined jinja variables shouldn't error out when the metadata is skipped.

I tracked it back to https://github.com/conda/conda-build/pull/5371, which I think is a good code improvement. In order to avoid these irrelevant errors for skipped recipes, we just abort after the first parsing pass.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda-build/blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks Michael! 🙏

If it is helpful, here is a generic example ( https://github.com/conda-forge/conda-smithy/issues/2011 )

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
@kenodegard could you please take a look as well? 🙂

File: news/5458-avoid-undefined-jinja-in-skipped [hunk 1]
```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* Allow undefined jinja variables when a particular metadata block evaluates as skipped
```
Comment (beeankha) [inline news/5458-avoid-undefined-jinja-in-skipped:7]:
```suggestion
* Allow undefined jinja variables when a particular metadata block evaluates as skipped. (#5458)
```

**Comment by kenodegard** [comment]
The tests are blocked by https://github.com/conda/conda/pull/14169

**Comment by jakirkham** [comment]
Thanks Bianca and Ken! 🙏

**Comment by kenodegard** [comment]
macOS tests are failing due to #5470

**Review by beeankha — APPROVED** [review]

**Review by kenodegard — APPROVED** [review]

**Comment by msarahan** [comment]
Thanks for the slog, @beeankha and @kenodegard

**Pull request merged**